pyop2/base.py

```diff
@@ -3415,7 +3415,7 @@ class JITModule(Cached):
     def _cache_key(cls, kernel, iterset, *args, **kwargs):
         counter = itertools.count()
         seen = defaultdict(lambda: next(counter))
-        key = ((id(dup_comm(iterset.comm)), ) + kernel._wrapper_cache_key_ + iterset._wrapper_cache_key_
+        key = (() + kernel._wrapper_cache_key_ + iterset._wrapper_cache_key_
```

Comment:
At least for compilation this is the first entry is ignored: https://github.com/OP2/PyOP2/blob/loopy-sprint/pyop2/compilation.py#L239

Reply:
Kaushik is right. This change is not the right thing to do. Check git blame for why the communicator is part of the key.

Reply:
https://github.com/OP2/PyOP2/commit/2e1a4b7a607e56602dc9aee19d06048e9d2b14ed

Reply:
I totally agree with both of you, this 'fix' is awful. The thing is is that somehow this change got the test passing for me. I know not how.

I created this PR so others could see what progress I'd made and discuss it. I should have made clear that I didn't want it merged.